flashlight/autograd/Functions.cpp

```diff
@@ -50,6 +50,7 @@ Variable operator+(const Variable& lhs, const Variable& rhs) {
   auto result = lhs.array() + rhs.array();
   auto gradFunc = [](std::vector<Variable>& inputs,
                      const Variable& grad_output) {
+    typeTrace("Operator+ BWD", grad_output.type());
```

Comment:
Just curious - I understand these are probably helpful for debugging; is the plan to remove these before merging?

Reply:
As you mentioned, it is useful for debugging. In particular, if in future a module gets revised and starts generating output in an undesired data type (e.g., due to a type promotion), this function can be used to quickly detect the reason for the performance hit. Otherwise, the program might be casting large tensors back and forth without us noticing. In terms of performance, It doesn't adversely impact the execution time, since it gets completely optimized away when the `TRACE_PRECISION` is not defined.